### PR TITLE
feat(cli): upgrade to TypeScript 4.3

### PR DESCRIPTION
**Pull request by kitsonk**
The PR updates to TypeScript 4.3.  This initial commit is the 4.3-beta for early visibility and testing.

TypeScript 4.3 planned release date is 25th May.

There is new TypeScript syntax in 4.3 that will require updates to swc to fully support (see: https://github.com/swc-project/swc/issues/1517).

### CODE REVIEW

**Comment by kitsonk** [comment]
swc has been updated while this was pending and should be compatible.

File: extensions/fetch/21_formdata.js [hunk 1]
```diff
@@ -332,37 +330,34 @@
...
       return this.#writeHeaders(headers);
-    };
-
-    /**
+    } /**
      * @param {string} field
      * @returns {void}
      */
-    #writeFieldHeaders = (field) => {
+
```
Comment (bartlomieju) [inline extensions/fetch/21_formdata.js:337]: Are these changes intentional? Looks like comment formatting breaks

**Review by bartlomieju — APPROVED** [review]

File: extensions/fetch/21_formdata.js (hunk 1 above)
Comment (kitsonk) [reply extensions/fetch/21_formdata.js:337]: hmm sort of `deno fmt` strangeness
